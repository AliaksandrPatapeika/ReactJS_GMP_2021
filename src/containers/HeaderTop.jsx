import '../less/containers/HeaderTop.less';

import React from 'react';

import AddMovieButton from '../components/AddMovieButton';
import Logo from '../components/Logo';

const HeaderTop = () => (
  <div className="headerTopContainer">
    <Logo />
    <AddMovieButton />
  </div>
);

export default HeaderTop;
