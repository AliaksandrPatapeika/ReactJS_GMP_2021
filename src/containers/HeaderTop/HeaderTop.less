@import "../../less/variables";

.headerTopContainer {
	display: flex;
	justify-content: space-between;

	.returnToMainButton {
		cursor: pointer;
		padding: 3px 6px;
		color: @red;
		font-size: 23px;
		background-color: transparent;

		&:hover,
		&:focus {
			color: @red-hover;
		}
	}

	.addMovieButton {
		cursor: pointer;
		width: 184px;
		height: 48px;
		border-radius: 4px;
		color: @red;
		font-size: 22px;
		font-family: @NetflixSans-Medium;
		background-color: @gray;
		opacity: 0.8;

		&:hover,
		&:focus {
			color: @white;
			background-color: @red-hover;
			opacity: 1;
		}
	}
}