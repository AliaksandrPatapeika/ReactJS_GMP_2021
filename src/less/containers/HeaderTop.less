.headerTopContainer {
	display: flex;
	justify-content: space-between;
	padding: 22px 60px 0 64px;
}