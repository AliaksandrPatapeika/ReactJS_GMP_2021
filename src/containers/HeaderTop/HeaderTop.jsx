import './HeaderTop.less';

import PropTypes from 'prop-types';
import React, {useEffect} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {useHistory} from 'react-router-dom';

import {closeMovieDetails, showModal} from '../../actions/movies';
import Button from '../../components/Button';
import Logo from '../../components/Logo';
import {getQueryString} from '../../utils';

const HeaderTop = ({hideButton}) => {
  const activeMovieDetailsMovie = useSelector((state) => state.movie.activeMovieDetailsMovie);
  const query = useSelector((state) => state.query);
  const history = useHistory();
  const dispatch = useDispatch();

  const onClickHandler = () => {
    dispatch(closeMovieDetails());
    history.push({
      pathname: '/search',
      search: getQueryString(query)
    });
  };

  const showAddMovieWindow = () => {
    dispatch(showModal('addMovie'));
  };

  useEffect(() => {
    if (activeMovieDetailsMovie) {
      document.getElementById('returnToMainButton').focus();
    }
  }, [activeMovieDetailsMovie]);

  return (
    <div className="headerTopContainer">
      <Logo />
      {!hideButton && (
      <>
        {activeMovieDetailsMovie ? (
          <Button
            id="returnToMainButton"
            className="returnToMainButton"
            onClick={onClickHandler}
          >
            <i className="fa fa-search fa-flip-horizontal" aria-hidden="true" />
          </Button>
        ) : (
          <Button
            className="addMovieButton"
            onClick={showAddMovieWindow}
            title="+ ADD MOVIE"
          />
        )}
      </>
      )}
    </div>
  );
};

HeaderTop.propTypes = {
  hideButton: PropTypes.bool
};

HeaderTop.defaultProps = {
  hideButton: false
};

export default HeaderTop;
