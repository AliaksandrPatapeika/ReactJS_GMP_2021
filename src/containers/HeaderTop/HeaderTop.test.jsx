import React from 'react';
import renderer from 'react-test-renderer';

import HeaderTop from './HeaderTop';

describe('HeaderTop', () => {
  test('renders correctly', () => {
    const headerTop = renderer
      .create(<HeaderTop />)
      .toJSON();

    expect(headerTop).toMatchSnapshot();
  });
});
