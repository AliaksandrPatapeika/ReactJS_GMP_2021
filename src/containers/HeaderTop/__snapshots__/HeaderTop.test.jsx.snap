// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTop renders correctly 1`] = `
<div
  className="headerTopContainer"
>
  <div
    className="logoContainer"
  >
    <span
      className="netflix"
    >
      netflix
    </span>
    <span
      className="roulette"
    >
      roulette
    </span>
  </div>
  <button
    className="addMovieButton"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    + ADD MOVIE
    
  </button>
</div>
`;
